File: pingdart-node/test/sdk.test.js
import PingDart from "../src/index.js";

describe("PingDart SDK Client", () => {
    const config = {
        apiKey: "test-key",
        databaseId: "test-db"
    };

    test("should initialize successfully", () => {
        const sdk = new PingDart(config);
        expect(sdk).toBeDefined();
        expect(sdk.apiKey).toBe(config.apiKey);
        expect(sdk.databaseId).toBe(config.databaseId);
    });

    test("should have all services mounted", () => {
        const sdk = new PingDart(config);
        expect(sdk.database).toBeDefined();
        expect(sdk.whatsapp).toBeDefined();
        expect(sdk.sms).toBeDefined();
        expect(sdk.email).toBeDefined();
        expect(sdk.ai).toBeDefined();
        expect(sdk.calls).toBeDefined();
        expect(sdk.storage).toBeDefined();
    });

    test("should throw error if apiKey is missing", () => {
        expect(() => new PingDart({})).toThrow("PingDart SDK: API Key is required");
    });
});

File: pingdart-dart/test/sdk_test.dart
import 'package:test/test.dart';
import 'package:pingdart_sdk/pingdart_sdk.dart';

void main() {
  group('PingDartSDK', () {
    final apiKey = 'test-key';
    final dbId = 'test-db';

    test('should initialize successfully', () {
      final sdk = PingDartSDK(apiKey: apiKey, databaseId: dbId);
      expect(sdk, isNotNull);
    });

    test('should have all services mounted', () {
      final sdk = PingDartSDK(apiKey: apiKey, databaseId: dbId);
      expect(sdk.database, isNotNull);
      expect(sdk.calls, isNotNull);
      expect(sdk.ai, isNotNull);
      expect(sdk.email, isNotNull);
      expect(sdk.sms, isNotNull);
      expect(sdk.storage, isNotNull);
    });
  });
}
